review/communication-react.api.md

```diff
@@ -0,0 +1,1828 @@
+    message: ChatMessage | SystemMessage | CustomMessage;
+    strings: MessageThreadStrings;
+    messageContainerStyle?: ComponentSlotStyle;
+    showDate?: boolean;
+};
+
+// @public (undocumented)
+export type MessageReadListener = (event: {
+    message: ChatMessage_2;
+    readBy: CommunicationUserKind;
+}) => void;
+
+// @public (undocumented)
+export type MessageReceivedListener = (event: {
+    message: ChatMessage_2;
+}) => void;
+
+// @public (undocumented)
+export type MessageSentListener = MessageReceivedListener;
+
+// @public (undocumented)
+export type MessageStatus = 'delivered' | 'sending' | 'seen' | 'failed';
+
+// @public
+export const MessageStatusIndicator: (props: MessageStatusIndicatorProps) => JSX.Element;
+
+// @public
+export interface MessageStatusIndicatorProps {
+    status?: MessageStatus;
+    strings?: MessageStatusIndicatorStrings;
+    styles?: BaseCustomStylesProps;
+}
+
+// @public
+export interface MessageStatusIndicatorStrings {
+    deliveredTooltipText: string;
+    failedToSendTooltipText: string;
+    seenTooltipText: string;
+    sendingTooltipText: string;
+}
+
+// @public
+export const MessageThread: (props: MessageThreadProps) => JSX.Element;
+
+// @public
+export type MessageThreadProps = {
+    userId: string;
+    messages: (ChatMessage | SystemMessage | CustomMessage)[];
+    styles?: MessageThreadStylesProps;
+    disableJumpToNewMessageButton?: boolean;
+    showMessageDate?: boolean;
+    showMessageStatus?: boolean;
+    numberOfChatMessagesToReload?: number;
+    onMessageSeen?: (messageId: string) => Promise<void>;
+    onRenderMessageStatus?: (messageStatusIndicatorProps: MessageStatusIndicatorProps) => JSX.Element | null;
+    onRenderAvatar?: OnRenderAvatarCallback;
+    onRenderJumpToNewMessageButton?: (newMessageButtonProps: JumpToNewMessageButtonProps) => JSX.Element;
+    onLoadPreviousChatMessages?: (messagesToLoad: number) => Promise<boolean>;
+    onRenderMessage?: (messageProps: MessageProps, defaultOnRender?: DefaultMessageRendererType) => JSX.Element;
+    strings?: Partial<MessageThreadStrings>;
+};
+
+// @public
+export interface MessageThreadStrings {
+    friday: string;
+    monday: string;
+    participantJoined: string;
+    participantLeft: string;
+    saturday: string;
+    sunday: string;
+    thursday: string;
+    tuesday: string;
+    wednesday: string;
+    yesterday: string;
+}
+
+// @public (undocumented)
+export interface MessageThreadStylesProps extends BaseCustomStylesProps {
+    chatContainer?: ComponentSlotStyle;
+    chatMessageContainer?: ComponentSlotStyle;
+    loadPreviousMessagesButtonContainer?: IStyle;
+    messageStatusContainer?: (mine: boolean) => IStyle;
+    newMessageButtonContainer?: IStyle;
+    systemMessageContainer?: ComponentSlotStyle;
+}
+
+// @public (undocumented)
+export type MessageTypes = 'chat' | 'system' | 'custom';
+
+// @public
+export const MicrophoneButton: (props: MicrophoneButtonProps) => JSX.Element;
```

Comment:
Fluents definitions are little different to this, they extend a react functional component - do we want to align with them?
e.g.
![image](https://user-images.githubusercontent.com/2684369/131418532-b7633673-de04-4066-a566-b4553beae178.png)

https://github.com/microsoft/fluentui/blob/master/packages/react/etc/react.api.md

Reply:
Their buttons are just parts of our implementation, I believe they keep it in a class style for legacy code support - for us, we can just align functional component all around the repo